go/teams/seitan_v2.go

```diff
@@ -0,0 +1,201 @@
+package teams
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/base64"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/go-crypto/ed25519"
+	"golang.org/x/net/context"
+
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+// Seitan tokens v2 have a '+' as the sixth character. We use this
+// to distinguish from email invite tokens (and team names).
+const seitanEncodedIKeyV2PlusOffset = 6
+
+// "Invite Key Version 2"
+type SeitanIKeyV2 string
+
+func GenerateIKeyV2() (ikey SeitanIKeyV2, err error) {
+	str, err := generateIKey(seitanEncodedIKeyV2PlusOffset)
+	if err != nil {
+		return ikey, err
+	}
+	return SeitanIKeyV2(str), err
+}
+
+// GenerateIKeyV2FromString safely creates SeitanIKey value from
+// plaintext string. Only format is checked - any 18-character token
+// with '+' character at position 6 can be "Invite Key". Alphabet is
+// not checked, as it is only a hint for token generation and it can
+// change over time, but we assume that token length stays the same.
+func GenerateIKeyV2FromString(token string) (ikey SeitanIKeyV2, err error) {
+	if len(token) != SeitanEncodedIKeyLength {
+		return ikey, fmt.Errorf("invalid token length: expected %d characters, got %d", SeitanEncodedIKeyLength, len(token))
+	}
+	if token[seitanEncodedIKeyV2PlusOffset] != '+' {
+		return ikey, fmt.Errorf("invalid token format: expected %dth character to be '+'", seitanEncodedIKeyV2PlusOffset+1)
+	}
+
+	return SeitanIKeyV2(strings.ToLower(token)), nil
+}
+
+func (ikey SeitanIKeyV2) String() string {
+	return strings.ToLower(string(ikey))
+}
+
+// "Stretched Invite Key"
+type SeitanSIKeyV2 [SeitanScryptKeylen]byte
+
+func (ikey SeitanIKeyV2) GenerateSIKey() (sikey SeitanSIKeyV2, err error) {
+	buf, err := generateSIKey(ikey.String())
+	if err != nil {
+		return sikey, err
+	}
+	copy(sikey[:], buf)
+	return sikey, nil
+}
+
+func (sikey SeitanSIKeyV2) GenerateTeamInviteID() (id SCTeamInviteID, err error) {
+	type InviteStagePayload struct {
+		Stage   string `codec:"stage" json:"stage"`
+		Version int    `codec:"version" json:"version"`
```

Comment:
let's make a data type for this since we now have two legit values for it.  `seitanVersion` or something with `seitanVersion1` and `seitanVersion2`.

Reply:
done